swap/src/monero.rs

```diff
@@ -19,6 +19,33 @@ use std::str::FromStr;
 
 pub const PICONERO_OFFSET: u64 = 1_000_000_000_000;
 
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Copy, Serialize, Deserialize)]
+pub enum Network {
+    Mainnet,
+    Stagenet,
+    Testnet,
+}
+
+impl From<monero::Network> for Network {
+    fn from(network: monero::Network) -> Self {
+        match network {
+            monero::Network::Mainnet => Self::Mainnet,
+            monero::Network::Stagenet => Self::Stagenet,
+            monero::Network::Testnet => Self::Testnet,
+        }
+    }
+}
+
+impl From<Network> for monero::Network {
+    fn from(network: Network) -> Self {
+        match network {
+            Network::Mainnet => monero::Network::Mainnet,
+            Network::Stagenet => monero::Network::Stagenet,
+            Network::Testnet => monero::Network::Testnet,
+        }
+    }
+}
```

Comment:
Instead of this indirection, would a serde module also help? If it is just for serde, using a module and `#[serde(with)]` will make things more ergonomic down the line because you don't need to convert back and forth.

Also, a module is easier to upstream. Alternatively, you can also upstream the `derive`. I am actually surprised it is not yet implemented.